ipld/retriever.go

```diff
@@ -239,6 +274,10 @@ func (rs *retrievalSession) doRequest(ctx context.Context, q *quadrant) {
 			// get the root node
 			nd, err := plugin.GetNode(ctx, rs.bget, root)
 			if err != nil {
+				rs.span.RecordError(err, trace.WithAttributes(
+					attribute.String("requesting-root", root.String()),
+					attribute.Int("root-id", i),
```

Comment:
```suggestion
					attribute.Int("root-index", i),
```
maybe a bit more descriptive?

Reply:
Not sure

Reply:
`id` i feel like corresponds to hash, and `i` is index